src/main/java/ee/carlrobert/codegpt/settings/state/OllamaSettingsState.java

```diff
@@ -0,0 +1,94 @@
+import ee.carlrobert.codegpt.completions.CompletionClientProvider;
+import ee.carlrobert.codegpt.completions.llama.LlamaCompletionModel;
+import ee.carlrobert.codegpt.credentials.OllamaCredentialsManager;
+import ee.carlrobert.codegpt.settings.state.llama.LlamaLocalSettings;
+import ee.carlrobert.codegpt.settings.state.llama.LlamaRemoteSettings;
+import ee.carlrobert.codegpt.settings.state.llama.LlamaRequestSettings;
+import ee.carlrobert.codegpt.settings.state.llama.ollama.OllamaRemoteSettings;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaModel;
+import ee.carlrobert.llm.client.ollama.completion.response.OllamaTagsResponse;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+//@State(name = "CodeGPT_OllamaSettings", storages = @Storage("CodeGPT_CodeGPT_OllamaSettings.xml"))
+public class OllamaSettingsState extends LlamaSettingsState<OllamaRemoteSettings> implements
+    PersistentStateComponent<OllamaSettingsState> {
+
+  public OllamaSettingsState() {
+    super(new OllamaRemoteSettings());
+  }
+
+  public OllamaSettingsState(boolean runLocalServer, LlamaLocalSettings localSettings,
+      OllamaRemoteSettings remoteSettings,
+      LlamaRequestSettings requestSettings) {
+    super(runLocalServer, localSettings, remoteSettings, requestSettings);
+  }
+
+  @Override
+  public OllamaSettingsState getState() {
+    return this;
+  }
+
+  @Override
+  public void loadState(@NotNull OllamaSettingsState state) {
+    XmlSerializerUtil.copyBean(state, this);
+  }
+
+  public static OllamaSettingsState getInstance() {
+    OllamaSettingsState service = ApplicationManager.getApplication()
+        .getService(OllamaSettingsState.class);
+    service.localSettings.setCredentialsManager(
+        new OllamaCredentialsManager(LlamaLocalSettings.CREDENTIALS_PREFIX));
+    service.remoteSettings.setCredentialsManager(
+        new OllamaCredentialsManager(LlamaRemoteSettings.CREDENTIALS_PREFIX));
+    return service;
+  }
+
+  @Transient
+  public boolean isModified(OllamaSettingsState settingsState) {
+    return localSettings.isModified(settingsState.getLocalSettings())
+        || remoteSettings.isModified(settingsState.getRemoteSettings())
+        || runLocalServer != settingsState.isRunLocalServer()
+        || requestSettings.isModified(settingsState.getRequestSettings());
+  }
+
+  public void apply(OllamaSettingsState settingsState) {
+    runLocalServer = settingsState.isRunLocalServer();
+    localSettings = settingsState.getLocalSettings();
+    remoteSettings = settingsState.getRemoteSettings();
+    requestSettings = settingsState.getRequestSettings();
+  }
+
+  @Transient
+  private List<OllamaModel> availableOllamaModels = null;
+
+  @Transient
+  public List<OllamaModel> getAvailableModels() {
+    if (availableOllamaModels == null) {
+      OllamaTagsResponse modelTags = CompletionClientProvider.getOllamaClient().getModelTags();
```

Comment:
I thought about using the `/api/tags` API Endpoint to get the available/downloaded models, maybe we could simply only request the list once when the user opens the settings